### PR TITLE
Start adding copy generated code snippet button and knob sections

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Update the color picker to match the figma designs
- Group Props by Type (required, optional)
- Add a copy code button on generated snippet, on playground pages
- Generated code is 30% height, live-example is 70% height on playground.
- Playground input values are now Monospaced
- Color picker has an error state on invalid value
- Number slider no longer shows arrows on hover
- If only a single section of properties on playground, do not show an accordion.

### CODE REVIEW

**Comment by ektasawant** [comment]
If we are going to match new figma design (below image) in this story then please consider following comments or else ignore.
Screenshot 2022-08-30 at 10 46 49 AM

1. Accordion with title should be there for optional props

**Comment by emclaug2** [comment]
> 1. Accordion with title should be there for optional props

There are no required props for `DrawerHeader`, `DrawerSubheader`, `DrawerNavGroup`, `DrawerNavItem`, or `DrawerFooter` so I put the group title as a generic "Properties" but I can update this to be "Optional Properties".

**Comment by emclaug2** [comment]
I've updated this PR to include the following changes:

1. Generated code is 30% height, live-example is 70% height on playground.
2. Playground input values are now Monospaced
3. Color picker has an error state on invalid value
4. Number slider no longer shows arrows on hover 
5. If only a single section of properties on playground, do not show an accordion.

**Comment by huayunh** [comment]
- [ ] 1. Drawer header playground, the "divider" prop shall be mono-spaced
- [ ] 2. Speaking of which, is Angular calling it "props"? Shall we spell it out as "properties"?
- [ ] 3. Are we going to split properties like "show icon" into the "others" bucket?
- [ ] 4. I noticed that the drawer nav item has an "add icon" property, while the drawer header playground is using the phrase "show icon"
- [ ] 5. Interesting... every time I make a change in those knobs, the code snippet scrolls to the top.

**Comment by emclaug2** [comment]
Hey @huayunh all your notes should be addressed now, except for the scroll-top behavior.  That's a weird effect; I'd like to log an issue for this & revisit it later.  I'm not sure why this is happening.

**Comment by huayunh** [comment]
Thanks Evan! I am reviewing this PR right now, but I noticed that whenever title text overflows, the drawer does interesting things... I expect the text to be either wrapped (header grows vertically) or truncated with an ellipsis. Not sure if this is an existing bug.

image

**Comment by huayunh** [comment]
- [ ] 1. Drawer component playground is now broken
- [ ] 2. Same for Drawer Body
- [ ] 3. Occasionally Chrome tries to warn me that this website is "deceptive"... this happened before I made my last comment. Right now if you go to the drawer and go to its playground and refresh, Chrome will give you a big red screen
- I just noticed that when drawer footer is closed, the divider disappears as well. The divider should stay, which is what React is doing right now. I will log an issue for this.

**Comment by emclaug2** [comment]
> Thanks Evan! I am still reviewing this PR right now, but I noticed that whenever title text overflows, the drawer does interesting things... I expect the text to be either wrapped (header grows vertically) or truncated with an ellipsis. Not sure if this is an existing bug.

This looks like a bug I introduced when I added the `width=250px` inline style to the Drawer.  I can remove it but the DrawerHeader example will be a little wider than before.  

Seems like NavItem example has the same issue.   Drawer `width` is a prop that's normally set on the DrawerLayout so just slapping a style tag on the root `<blui-drawer>` didn't work as well as I'd hoped.

**Comment by emclaug2** [comment]
> * [ ]  1. Drawer component playground is now broken

This PR was opened before the DrawerPlayground was added so it won't exist here.


> * [ ]  2. Same for Drawer Body

DrawerBody needs some help; I'll update this PR so that the Examples & Playground page is omitted from the view.

> * [ ]  3. Occasionally Chrome tries to warn me that this website is "deceptive"... this happened before I made my last comment. Right now if you go to the drawer and go to its playground and refresh, Chrome will give you a big red screen

I've seen this before but haven't looked into it yet; I'm guessing this something that'll affect all of our firebase deployments.  Maybe this could be a spike for next sprint.

**Comment by huayunh** [comment]
Is this story still "**start** adding knobs", as indicated by your PR title, or am I expected to review your branch with designer eyes?

**Comment by emclaug2** [comment]
> Is this story still "**start** adding knobs", as indicated by your PR title, or am I expected to review your branch with designer eyes?

Bring on the designer eyes; as long as there's nothing significantly wrong I can update the site as part of this PR.

**Comment by jeffvg** [comment]
at this point for this PR, should all drawer components playground be up to date?
The drawer header themed background color: primary, secondary & warn doesn't seem to be working right now but maybe its for another PR.

For the drawer foot playground maybe reword the "Nav item" in example to match what the knob states?
- Show divider under subheader with the example to have "subheader item" rather than "Nav item"

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
11. texts in the table of "API Docs" are not currently following our "monospaced" rules
12. Quoted texts in API docs also need their styles adjusted
13. Interesting... I didn't know that `title` is not a required property in a drawer nav item.

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
image

**Comment by huayunh** [comment]
My comments above are intended to be a comprehensive review of this PR. Some may be out of the scope for this PR, or even this release. Please feel free to write issues / stories, or punt them to another PR to keep things moving at your preferred pace.

**Comment by huayunh** [comment]
18: According to design, boolean types need to show their type names too — divider **: boolean**

(Screenshot from Figma)
image

**Comment by emclaug2** [comment]
I'm going to address the comments in the reverse order.

In summary - anything API, link, or description related I've deferred to a separate issue since they can be grouped together. 

17 - Fixed
16 - Issue made since it'll take some time.
15 - I'll keep this as string because DrawerLayout `variant` has so many types that it won't fix on a single line.  
14 - Fixed
13 - Yep required props are a rare thing in angular-land
12 - Issue made
11 - Issue made
10 - Fixed 
9 -  Issue made
8 - ng-material doesn't have multi-line hints gracefully, I've patched this but it's a manual process & this issue might come up again whenever knob hints are longer.  Just something to keep an eye out for.
7 - Issue made
6 - Fixed
5 - Issue made
4 - Fixed
3 - Fixed
2 - Fixed
1 - Designs are missing a border? I'm pretending there is a border there so all items are aligned.

**Review by huayunh — APPROVED** [review]
You may want to auto-sort your knobs alphabetically. Otherwise lgtm (didn't check the code)
